### PR TITLE
Move the access control ownership and initialization to Server.

### LINKED ISSUE #10251: Implement simple access control cluster persistence/storage
- design storage format
- implement load/save functionality
- ensure system is robust (data integrity etc.)
- lean and fast also important (in that order)
- defer extensions for now, but keep them in mind (#10252)

### PR DESCRIPTION
**Pull request by harimau-qirex**
#### Problem
The simple ACL storage added in #14253 is only supported by Linux and MacOS. By moving the ownership of the AccessControl object to Server and using the platform storage delegate that it owns, all platforms that use Server will get ACL persistence.
Fixes #10251.

#### Change overview
* Move the ownership of the AccessControl object from the ACL cluster to Server.
* Remove the ACL storage delegate from the AppMain.cpp used by Linux and MacOS.
* Initializing ACL without storage or stored ACL data is no longer considered a failure.

#### Testing
* Manually tested on MacOS by commissioning all-clusters-app, writing ACL entries, then restarting all-clusters-app and verifying the entries were still present.

### CODE REVIEW

**Comment by bzbarsky-apple** [comment]
Just as a note, this is adding several KB or RAM use.  We should probably have  followup issue to reduce that...

**Comment by mlepage-google** [comment]
We do already have issue #14450 for general Access Control performance (all aspects), @bzbarsky-apple is that sufficient or do we also want a more targeted/specific issue?

File: src/access/examples/ExampleAccessControlDelegate.cpp [hunk 1]
```diff
@@ -1069,12 +1069,13 @@ class AccessControlDelegate : public AccessControl::Delegate
         CHIP_ERROR err = LoadFromFlash();
         if (err != CHIP_NO_ERROR)
         {
+            ChipLogDetail(DataManagement, "Unable to load stored ACL entries; using empty list instead");
```
Comment (mlepage-google) [inline src/access/examples/ExampleAccessControlDelegate.cpp:1072]: I've been switching to this kind of message for logs.
```suggestion
            ChipLogDetail(DataManagement, "AccessControl: unable to load stored ACL entries; using empty list instead");
```

File: src/app/clusters/access-control-server/access-control-server.cpp [hunk 2]
```diff
@@ -501,16 +501,9 @@ CHIP_ERROR AccessControlAttribute::WriteExtension(AttributeValueDecoder & aDecod
 
 AccessControlAttribute gAttribute;
 
-AccessControl gAccessControl(Examples::GetAccessControlDelegate());
```
Comment (mlepage-google) [inline src/app/clusters/access-control-server/access-control-server.cpp:504]: Can the include be removed now?
`#include <access/examples/ExampleAccessControlDelegate.h>`

**Comment by harimau-qirex** [comment]
I'm currently looking into the cause of the size differences to see if it's something not already covered by #14450.

**Comment by bzbarsky-apple** [comment]
> is that sufficient or do we also want a more targeted/specific issue?

@mlepage-google I would prefer a more targeted issue, for this and the increase from https://github.com/project-chip/connectedhomeip/pull/14657 (which might be the same increase?).

**Review by tcarmelveilleux — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

**Comment by harimau-qirex** [comment]
My investigations showed that the size increases for the apps I could check were generally due to the ACL code. It was previously removed from the build.

**Comment by bzbarsky-apple** [comment]
>  It was previously removed from the build.

How could it have been removed from the build when it was used in the acl cluster?

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**